src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -454,8 +454,12 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 		}
 
 		const completion = suggestion.item.completion;
-		const completionText = completion.label;
-
+		let completionText = completion.label;
+		if ((completion.isDirectory || completion.isFile) && completionText.includes(' ')) {
+			// Wrap files or folders in quotes if they contain spaces so they're
+			// valid paths
+			completionText = `"${completionText}"`;
+		}
```

Comment:
This is more complicated than this, paths can have quotes themselves but what you'll want to do on most shells is escape the spaces instead, so `hello world` becomes `hello\ world`. This is especially important for directories due to https://github.com/microsoft/vscode/issues/234621